tracee-rules/signature.go

```diff
@@ -45,52 +45,55 @@ func getSignatures(rulesDir string, rules []string) ([]types.Signature, error) {
 }
 
 func findGoSigs(dir string) ([]types.Signature, error) {
-	files, err := ioutil.ReadDir(dir)
-	if err != nil {
-		return nil, fmt.Errorf("error reading plugins directory %s: %v", dir, err)
-	}
 	var res []types.Signature
-	for _, file := range files {
-		if filepath.Ext(file.Name()) != ".so" {
-			continue
+	err := filepath.Walk(dir, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return fmt.Errorf("error opening plugin %s: %v", info.Name(), err)
 		}
-		p, err := plugin.Open(filepath.Join(dir, file.Name()))
+		if filepath.Ext(info.Name()) != ".so" || info.IsDir() == true {
+			return filepath.SkipDir
+		}
+		p, err := plugin.Open(filepath.Join(path, info.Name()))
 		if err != nil {
-			log.Printf("error opening plugin %s: %v", file.Name(), err)
-			continue
+			log.Printf("error opening plugin %s: %v", info.Name(), err)
+			return filepath.SkipDir
 		}
 		export, err := p.Lookup("ExportedSignatures")
 		if err != nil {
-			log.Printf("missing Export symbol in plugin %s", file.Name())
-			continue
+			log.Printf("missing Export symbol in plugin %s", info.Name())
+			return filepath.SkipDir
 		}
 		sigs := *export.(*[]types.Signature)
 		res = append(res, sigs...)
+		return filepath.SkipDir
```

Comment:
is this necessary?

Reply:
No, it's not. I've changed it to return `nil` instead.